quizcon/main/tests/test_models.py

```diff
@@ -3,13 +3,60 @@
     CourseTestMixin, QuizFactory, QuestionFactory, QuestionResponseFactory,
     QuizSubmissionFactory
 )
+from quizcon.main.models import Quiz
 
 
 class BasicModelTest(TestCase):
     def test_dummy(self):
         self.assertTrue(True)
 
 
+class QuizCloneTest(CourseTestMixin, TestCase):
+    def setUp(self):
+        self.setup_course()
+
+    def test_clone(self):
+        q = QuizFactory(course=self.course, title='cloned quiz')
+        q.question = QuestionFactory(quiz=q)
+
+        cloned_pk = q.clone().pk
+        c = Quiz.objects.get(pk=cloned_pk)
+
+        self.assertNotEqual(q.pk, c.pk)
+        self.assertEqual(q.title, 'cloned quiz')
+        self.assertEqual(c.title, 'cloned quiz')
+
+        c.title = 'new title'
+        q.save()
+        c.save()
+
+        q.refresh_from_db()
+        c.refresh_from_db()
+
+        self.assertNotEqual(q.pk, c.pk)
+        self.assertEqual(q.title, 'cloned quiz')
+        self.assertEqual(c.title, 'new title')
+        self.assertEqual(q.scoring_scheme, c.scoring_scheme)
+
+        c.scoring_scheme = 0
+        c.save()
+        c.refresh_from_db()
+        self.assertEqual(c.scoring_scheme, 0)
+        self.assertNotEqual(q.scoring_scheme, c.scoring_scheme)
```

Comment:
This is super super nit picky: once you've asserted that they're different pk's, I don't think you need to test that a change to one will change the other. At that point, it's not necessarily testing your cloning code, but rather it's testing that Django is updating the object. You don't have to change anything, but I'm mentioning because it can be kind of a trap when you're writing tests.

Reply:
Yes I was kind of thinking the same while going through it. I'll make those changes!